h5py/_hl/dataset.py

```diff
@@ -53,10 +53,10 @@ def readtime_dtype(basetype, names):
 
 
 def make_new_dset(parent, shape=None, dtype=None, data=None,
-                 chunks=None, compression=None, shuffle=None,
-                    fletcher32=None, maxshape=None, compression_opts=None,
+                  chunks=None, compression=None, shuffle=None,
+                  fletcher32=None, maxshape=None, compression_opts=None,
                   fillvalue=None, scaleoffset=None, track_times=None,
-                  external=None):
+                  track_order=None, external=None):
```

Comment:
Please do not re-order args.  There are technically positional and there maybe someone who is making the choice to pass in 14 positional arguments that this would break!